src/capture_RPiHQ.cpp

```diff
@@ -251,294 +266,253 @@ void writeToLog(int val)
-		roi = "--mode 3 ";
+		command += " --mode 3";
 	}
-	// Check if binning 2x2 is selected
 	else if (bin==2) 	{
-		// Select binning 2x2 (2028 x 1520 pixels)
-		roi = "--mode 2  --width 2028 --height 1520 ";
+		command += " --mode 2  --width 2028 --height 1520";
 	}
-	// Check if binning 3x3 is selected
 	else 	{
-		// Select binning 4x4 (1012 x 760 pixels)
-		roi = "--mode 4 --width 1012 --height 760 ";
+		command += " --mode 4 --width 1012 --height 760";
 	}
 
-	// Append binning window
-	command += roi;
-
 	if (asiExposure < 1)
 	{
 		asiExposure = 1;
 	}
-
-    // https://www.raspberrypi.org/documentation/raspbian/applications/camera.md : HQ (IMX477) 	200000000 (i.e. 200s)
-	if (asiExposure > 200000000)
+	else if (asiExposure > 200000000)
 	{
+		// https://www.raspberrypi.org/documentation/raspbian/applications/camera.md : HQ (IMX477) 	200000000 (i.e. 200s)
 		asiExposure = 200000000;
 	}
 
-	// Exposure time
-	string shutter;
-
 	// Check if automatic determined exposure time is selected
 
 	if (asiAutoExposure)
 	{
 		if (myModeMeanSetting.mode_mean) {
 			ss.str("");
 			ss << myRaspistillSetting.shutter_us;
-			shutter = "--exposure off --shutter " + ss.str() + " ";
+			command += " --exposure off";
+			command += " --shutter " + ss.str();
 		} else {
-			shutter = "--exposure auto ";
+			command += " --exposure auto";
 		}
 	}
 	// Set exposure time
 	else if (asiExposure)
 	{
 		ss.str("");
 		ss << asiExposure;
-		shutter = "--exposure off --shutter " + ss.str() + " ";
+		command += " --exposure off";
+		command += " --shutter " + ss.str();
 	}
 
-	// Add exposure time setting to raspistill command string
-	command += shutter;
-
 	if (asiAutoFocus)
 	{
-		command += "--focus ";
+		command += " --focus";
 	}
 
-	// Anolog Gain
-	string gain;
-
 	// Check if auto gain is selected
 	if (asiAutoGain)
 	{
 		if (myModeMeanSetting.mode_mean) {
 			ss.str("");
 			ss << myRaspistillSetting.analoggain;
-			gain = "--analoggain " + ss.str() + " ";
+			command += " --analoggain " + ss.str();
 
 			if (myRaspistillSetting.digitalgain > 1.0) {
 				ss.str("");
 				ss << myRaspistillSetting.digitalgain;
-				gain = gain + "--digitalgain " + ss.str() + " ";
+				command += " --digitalgain " + ss.str();
 			}
 		}
 		else {
-			// Set analog gain to 1
-			gain = "--analoggain 1 ";
+			command += " --analoggain 1";	// 1 effectively makes it autogain
 		}
 	}
-	// Set manual analog gain setting
-	else if (asiGain) {
-		if (asiGain < 1) {
-			asiGain = 1;
+	else {							// Is manual gain
+		if (asiGain < 1.0) {
+			asiGain = 1.0;
 		}
-
-		if (asiGain > 16) {
-			asiGain = 16;
+		else if (asiGain > 16.0) {
+			asiGain = 16.0;
 		}
-
 		ss.str("");
 		ss << asiGain;
-		gain = "--analoggain " + ss.str() + " ";
+		command += " --analoggain " + ss.str();
 	}
 
-	// Add gain setting to raspistill command string
-	command += gain;
-
-  // Add exif information to raspistill command string
 	if (myModeMeanSetting.mode_mean) {
```

Comment:
This was a test code, because I get no useable EXIF Infos from raspistill. remove?

Reply:
I like the idea of adding EXIF information, and think we should add a lot more.  However, I'm not sure what EXIF fields we can add to, but it would be nice to add the person's name, the camera model, exposure, temperature (for cameras that support it), gain, and other settings.